src/Access/LDAPClient.cpp

```diff
@@ -26,30 +26,30 @@ namespace ErrorCodes
     extern const int LDAP_ERROR;
 }
 
-void LDAPClient::SearchParams::combineHash(std::size_t & seed) const
+void LDAPClient::SearchParams::updateHash(SipHash & hash) const
 {
-    boost::hash_combine(seed, base_dn);
-    boost::hash_combine(seed, static_cast<int>(scope));
-    boost::hash_combine(seed, search_filter);
-    boost::hash_combine(seed, attribute);
+    hash.update(base_dn);
+    hash.update(static_cast<int>(scope));
+    hash.update(search_filter);
```

Comment:
This is wrong, as it can return identical hashes for
('a', 'bc')
and
('ab', 'c')

String length should be mixed as well.

Reply:
fixed